### PR TITLE
Zen0bit refactoring

### PR DESCRIPTION
**Pull request by ivan-hc**

### CODE REVIEW

**Comment by ivan-hc** [comment]
@Samueru-sama @zen0bit @nazdridoy 

this PR includes all changes made by @zen0bit while refactoring

you can start testing this by removing AM
```
am -R am
```
and installing it using these commands
```
wget -q https://raw.githubusercontent.com/ivan-hc/AM/dev/INSTALL && chmod a+x ./INSTALL && sudo ./INSTALL
am --devmode-enable
am -s
```

see you next. Bye.

**Comment by zen0bit** [comment]
Why I am cleaning history?

If you merge with "Merge commit" again... 😢

You should merge ~~only commits~~, I already shown how to do that properly...

**Comment by ivan-hc** [comment]
> Why I am cleaning history?
> 
> If you merge with "Merge commit" again... 😢

@zen0bit what's up now?

by watching quickly your code on your fork and my forks I can see in the module install.am that the function "_install" its twiced. Not my mistake but yours.

I've not tested yet your changes (but if there is such an error... better if I shut up)

> You should merge **only commits**, I already shown how to do that properly...

I kindly asked you to do one thing: drag/paste. Stop. Why involve me in these operations? You are the one who is changing the code to give it "style". To give you space and time, I can only do two things in this repository and for this project in the meantime:

- help my users fix bugs in installation scripts;
- fix bugs in the modules and, if necessary, report it to you so that you can implement such changes.

All this translates into an abrupt stop to the changes that I personally, with my ideas, would like to make to my CLI... and which I cannot implement, to give you space and time so that you can introduce these corrections... precisely because, from your point of view, my way of writing code sucks.

That said, make the changes you see fit and do them well. My role here is only to notify you if I have changed something in "main", as there are users who have issues and ask for help, in this and other repositories managed by me. That's my priority, the users! Those who use my code daily! You're changing it, it's different.

And since what I have produced, by my own hand, now works in "main", what you change is your problem, I can only help you improve it, pointing out emergency changes that I had to make to the code in the meantime, so as not to make you lose track with "main". It's all too much what I'm doing for you now. Don't ask me anything else. Just do a good job.

See you.

**Comment by zen0bit** [comment]
Don't care about my refactoring.
Push anything you want or is needed!
Since you don't want work with my refactored code, don' see any reason why wait for me...

**Comment by ivan-hc** [comment]
> Don't care about my refactoring. Push anything you want or is needed! Since you don't want work with my refactored code, don' see any reason why wait for me...

@zen0bit You've been trying to introduce a full refactoring for months, and now I'm giving you the opportunity to do it. It's not a matter of not wanting to work with your code, otherwise I wouldn't have given you 2 weeks to do what you were doing. But realize that "AM" already works well, and it is not right to slow down the growth of such a project with different code that might not work. The ultimate goal of "AM" is to be used by people who need it, and it must work well.

But I must say that in your work here, despite the good work done, there are many writing errors in the code, I just read some modules and have not tested them yet, in "install.am" the "_install" function is double, in "help.am" there is a line where the name of the CLI and its path is written several times... now I don't remember other details I looked at, but... reading this code, I said to myself " But what has @zen0bit  been doing all this time?".

Don't be offended, believe me when I tell you that I appreciated, for example, how you merged the functions in "install.am" (even if I don't know if they will work), but if you want to learn to write code you have to do it little by little. You must first test whether your code works as it should, since Shellcheck only takes into account the syntax, not the functionality of the code itself.

To give you an example, "INSTALL", if you run the script by putting the "$currentuser" variable in quotes, you have access to /opt/am only as root, but if you remove the quotes, you will have access to the aforementioned directory without requiring privileges. Yet Shellcheck says that's wrong.

Having said that, we are all good at writing code following syntax rules, but in practice the important thing is that it works.

@nazdridoy 's code itself, regarding the "about" option, brought errors in Shellcheck, yet it worked. That's what you're missing, @zen0bit , a bit of anarchy.

You've been writing code for months without checking that it works, and only because "of course it should work, if Shellcheck says so".

To recap:
- Have you converted everything into functions? You need to remove duplicates!
- Have you written the functions? Test them, even at the cost of losing data on your system. You are the one who is sneaking into other people's PCs with your code and if you cause damage, it is your responsibility (see "XZ"). Assume the risks before posting.
- Have you finished the tests? Just let me know.
- I ask you if I can merge, yes or no? Just answer yes or no.

I've been telling you this for months. I told you that I created this project "for fun", but you are playing with it more than me. 

The only difference between you and me is that if I lose the game I start the level all over again... while you instead break the console hoping to win (and the console is "AM").
